### PR TITLE
Add org.webcamoid.Webcamoid

### PR DESCRIPTION
**Pull request by hipersayanX**
### Please confirm your submission meets all the criteria

- [:heavy_check_mark: ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [:heavy_check_mark: ] My pull request follows the instructions at [App Submission][submission].
- [:heavy_check_mark: ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [:heavy_check_mark: ] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [:heavy_check_mark: ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [:heavy_check_mark: ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [:heavy_check_mark: ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.webcamoid.Webcamoid.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+app-id: org.webcamoid.Webcamoid
```
Comment (hfiguiere) [inline org.webcamoid.Webcamoid.yml:1]: `webcamoid.org` doesnt resolve. Is this valid?

I see it's the one used upstream, but it has to be a valid domain under the project control.

File: org.webcamoid.Webcamoid.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: webcamoid
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
```
Comment (hfiguiere) [inline org.webcamoid.Webcamoid.yml:9]:
```suggestion
  - --socket=fallback-x11
```

with proper wayland support

File: org.webcamoid.Webcamoid.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --filesystem=host
```
Comment (hfiguiere) [inline org.webcamoid.Webcamoid.yml:12]: for a webcam application that's a very broad permission. I'd see `xdg-pictures` as a suitable one.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
You probably also need `rename-icon` `rename-desktop-file` and `rename-appdata-file`

**Comment by hfiguiere** [comment]
bot, build org.webcamoid.Webcamoid

File: org.webcamoid.Webcamoid.yml (hunk 1 above)
Comment (hfiguiere) [reply org.webcamoid.Webcamoid.yml:1]: I would see `io.github.webcamoid.Webcamoid` being a proper id meeting the requirements.

Comment (hfiguiere) [reply org.webcamoid.Webcamoid.yml:1]: To quote the documentation:

> Each application should have a unique application ID, following the standard reverse-DNS schema. See [the Flatpak documentation](http://docs.flatpak.org/en/latest/conventions.html#application-ids) for more information on this. The Application ID should be a real URL of a domain that the app author has control over or where their app is hosted.

You can disagree. But that's the policy.

File: org.webcamoid.Webcamoid.yml (hunk 3 above)
Comment (hfiguiere) [reply org.webcamoid.Webcamoid.yml:12]: not sure why it would need desktop or download, but it's way better than `host`

**Comment by hfiguiere** [comment]
build failed because the appdata/appstream file hasn't been found.

**Comment by hfiguiere** [comment]
> You probably also need `rename-icon` `rename-desktop-file` and `rename-appdata-file`

As I said this is why the build fails.

**Comment by hfiguiere** [comment]
> You probably also need `rename-icon` `rename-desktop-file` and `rename-appdata-file`

As I said this is why the build fails.

File: org.webcamoid.Webcamoid.yml (hunk 1 above)
Comment (barthalion) [reply org.webcamoid.Webcamoid.yml:1]: This is also the future way you will be able to verify that you really are the maintainer of the app, and get a checkmark on the website.

File: io.github.webcamoid.Webcamoid.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+app-id: io.github.webcamoid.Webcamoid
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: webcamoid
+rename-icon: webcamoid
+rename-appdata-file: io.github.webcamoid.metainfo.xml
```
Comment (hfiguiere) [inline io.github.webcamoid.Webcamoid.yml:7]: you could name the file properly in the repository, and you'd not need to do that.

File: io.github.webcamoid.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hfiguiere) [inline io.github.webcamoid.metainfo.xml:1]: this file isn't referenced anywhere, let alone installed.

File: io.github.webcamoid.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    <summary xml:lang="fr">Prenez des photos et enregistrer des vidéos avec votre webcam</summary>
+    <summary xml:lang="gl">Facer fotos e gravar vídeos coa súa cámara web</summary>
+    <summary xml:lang="it">Scatta foto e registrare video con la tua webcam</summary>
+    <summary xml:lang="ja">ウェブカメラで写真や記録ビデオを撮影</summary>
+    <summary xml:lang="ko">웹캠으로 사진과 기록 비디오를 촬영</summary>
+    <summary xml:lang="pt">Tirar fotos e gravar vídeos com sua webcam</summary>
+    <summary xml:lang="ru">Возьмите фотографии и записывать видео с веб-камеры</summary>
+    <summary xml:lang="zh_CN">拍摄照片和录制视频与您的摄像头</summary>
+    <summary xml:lang="zh_TW">拍攝照片和錄製視頻與您的攝像頭</summary>
+    <launchable type="desktop-id">io.github.webcamoid.Webcamoid</launchable>
```
Comment (hfiguiere) [inline io.github.webcamoid.metainfo.xml:19]:
```suggestion
    <launchable type="desktop-id">io.github.webcamoid.Webcamoid.desktop</launchable>
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by barthalion** [comment]
bot, build org.webcamoid.Webcamoid

**Comment by barthalion** [comment]
bot, build io.github.webcamoid.Webcamoid

**Comment by hfiguiere** [comment]
as I mentionned the appdata file is never installed.

File: io.github.webcamoid.metainfo.xml (hunk 5 above)
Comment (barthalion) [reply io.github.webcamoid.metainfo.xml:1]: To clarify: you need to add `install -Dm644 $FLATPAK_ID.metainfo.xml /app/share/metainfo/$FLATPAK_ID.metainfo.xml` to `build-commands` of some module, and reference this file in the `sources` array. This is the last change needed to make build, probably.

File: io.github.webcamoid.Webcamoid.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+app-id: io.github.webcamoid.Webcamoid
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: webcamoid
+rename-icon: webcamoid
+rename-appdata-file: io.github.webcamoid.Webcamoid.metainfo.xml
```
Comment (hfiguiere) [inline io.github.webcamoid.Webcamoid.yml:7]:
```suggestion
```

File: io.github.webcamoid.Webcamoid.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    post-install:
+      - install -Dm644 io.github.webcamoid.Webcamoid.metainfo.xml $FLATPAK_DEST/share/metainfo/io.github.webcamoid.Webcamoid.metainfo.xml
+  - name: webcamoid
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: git
+        url: https://github.com/webcamoid/webcamoid.git
+        tag: 9.0.0
```
Comment (hfiguiere) [inline io.github.webcamoid.Webcamoid.yml:35]:
```suggestion
  - name: webcamoid
    buildsystem: cmake-ninja
    config-opts:
      - -DCMAKE_BUILD_TYPE=Release
    post-install:
      - install -Dm644 io.github.webcamoid.Webcamoid.metainfo.xml -t $FLATPAK_DEST/share/metainfo
    sources:
      - type: git
        url: https://github.com/webcamoid/webcamoid.git
        tag: 9.0.0
      - type: file
        path: io.github.webcamoid.Webcamoid.metainfo.xml
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
the name of the metainfo file is still incorrect.

**Comment by hipersayanX** [comment]
> the name of the metainfo file is still incorrect.

I don't understand, how it should be named?

File: io.github.webcamoid.Webcamoid.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    post-install:
+      - install -Dm644 io.github.webcamoid.Webcamoid.metainfo.xml -t $FLATPAK_DEST/share/metainfo
+    sources:
+      - type: git
+        url: https://github.com/webcamoid/webcamoid.git
+        tag: 9.0.0
+      - type: file
+        path: io.github.webcamoid.Webcamoid.metainfo.xml
```
Comment (hfiguiere) [inline io.github.webcamoid.Webcamoid.yml:29]: > > the name of the metainfo file is still incorrect.
> 
> I don't understand, how it should be named?

here, you got the name right. unlike for the actual file in the repo.

Comment (hipersayanX) [reply io.github.webcamoid.Webcamoid.yml:29]: I got it, I forgot to rename the file.

**Comment by barthalion** [comment]
bot, build io.github.webcamoid.Webcamoid

**Comment by barthalion** [comment]
Appdata validation failed:

```
builddir/files/share/appdata/io.github.webcamoid.Webcamoid.appdata.xml: FAILED:
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://webcamoid.github.io/theme/images/screenshots/Main.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://webcamoid.github.io/theme/images/screenshots/Effects.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://webcamoid.github.io/theme/images/screenshots/DesktopCapture.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://webcamoid.github.io/theme/images/screenshots/VirtualCamera.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://webcamoid.github.io/theme/images/screenshots/Recording.png]
• tag-missing           : <release> required
```

**Comment by hfiguiere** [comment]
bot, build io.github.webcamoid.Webcamoid

**Comment by barthalion** [comment]
bot, build io.github.webcamoid.Webcamoid

**Comment by barthalion** [comment]
The virtual camera feature should be probably hidden altogether in Flatpak version, but that's not a blocker.

/merge

**Comment by hipersayanX** [comment]
@barthalion there is no way to have access to the virtual camera from flatpak? maybe giving some permissions? using portals?

**Comment by barthalion** [comment]
The app seems to download some installer which expects kernel headers and GCC to be available. There's no way to make that work under Flatpak. The message could be changed to point to some outside website with steps describing what to do directly on the host.

**Comment by beedell-roke** [comment]
> https://github.com/flathub/flathub/pull/2920#issuecomment-1083539361

@hipersayanX, yeah: https://github.com/flathub/com.obsproject.Studio/issues/91#issuecomment-822017770.

**Pull request closed without merging**